base/doc/ltnews37.tex

```diff
@@ -149,6 +149,29 @@ \section{New or improved commands}
 %
 \githubissue{xxx}
 
+\subsection{Extension of the \cs{label} command}
+
+Up to now the \cs{label} command stored in standard \LaTeX{} into the
+\cs{newlabel} command in the \texttt{.aux} file two data: 
+\cs{@currentlabel}, which normally contains the state of the current 
+counter, and \cs{thepage} for the current page number.  \pkg{hyperref} and 
+\pkg{nameref} patch the \cs{label} command to store five data: 
+Additionally they save \cs{@currentlabelname}, which normally contains the 
+current title text and can be retrieved with \cs{nameref} and 
+\cs{@currentHref} which is the name of the destination needed to create an 
+active link. The fifth data container is currently only used if external 
+references are loaded with the \pkg{xr-hyper} package. 
+
+Starting with this release the number of arguments have be unified. 
+\cs{label} now writes a
+\cs{newlabel} command containing five data: \cs{@currentlabel},
+\cs{thepage}, \cs{@currentlabelname}, \cs{@currentHref}. The fifth 
+container is reserved for the kernel and contains data needed for the 
+tagged PDF project. 
+
+Additionally a hook with the name \texttt{label} has been added. It is 
+executed before the writing to the \texttt{.aux} file. Code using
+the hook can refer to the label argument with \cs{label@name}. 
```

Comment:
if \@currentlabelname  etc why  \label@name  and not  \@labelargument, ie the @ always in the same position?

Reply:
from a general perspective I wonder if there shouldn't actually be 6 arguments (the last 2 key/value lists)
with 

 - arg5 reserved for packages to pass stuff around
 - arg6 reserved for the kernel to pass extra stuff around

shold probably be discussed elsewhere, not in this pull.

Reply:
> if @currentlabelname etc why \label@name and not @labelargument, ie the @ always in the same position?

`\label@name` is the command currently used by nameref. But `\@labelargument would be fine with me. 

> 6 arguments (the last 2 key/value lists)

perhaps, but at first the patches from hyperref and nameref should disappear.

Reply:
If changing to 6 args is too difficult, arg 5 could contain two groups, one for holding the user-information (in a well-defined and consistent format) and the other for the kernel.

Reply:
As discussed yesterday, we can do with one argument as long as we define that datastructure in a way that packages (or  users) can add to it in a controlled way. But if not we can indeed use 2 brace groups to separate them if that seems advisable in the end.

Reply:
argument ==> (data item)